edge-hub/src/Microsoft.Azure.Devices.Edge.Hub.Core/device/DeviceMessageHandler.cs

```diff
@@ -404,10 +404,16 @@ public async Task SendMessageAsync(IMessage message, string input)
             // The messages need to be processed in order.
             using (await this.serializeMessagesLock.LockAsync())
             {
+                var taskCompletionSource = new TaskCompletionSource<bool>();
+                if (!this.IsActive)
+                {
+                    taskCompletionSource.SetException(new EdgeHubIOException("Cannot send message. The communication channel is not active anymore."));
+                    await taskCompletionSource.Task;
```

Comment:
We should probably return from here, right?

Reply:
Also, do you need the TCS here?

Reply:
it is a convoluted way to throw an exception (and so return), mechanically copied from few lines below - where it makes more sense because those lines not necessarily throw - and you are right, I can throw it directly without the completion source.